src/body/body.rs

```diff
@@ -323,7 +323,12 @@ impl Body {
                     ping.record_data(bytes.len());
                     Poll::Ready(Some(Ok(bytes)))
                 }
-                Some(Err(e)) => Poll::Ready(Some(Err(crate::Error::new_body(e)))),
+                Some(Err(e)) => match e.reason() {
+                    // These reasons should cause stop of body reading, but nor fail it.
+                    // The same logic as for `AsyncRead for H2Upgraded` is applied here.
```

Comment:
I'm referring to this line here: https://github.com/hyperium/hyper/blob/a45d5d5a04369f93334fc893875d8d1a49054e04/src/proto/h2/mod.rs#L296

Both of them call the same operation (`self.recv_stream.poll_data(cx)`) and seems that logic should be the same. But we can make it more specific.